src/hooks.server.ts

```diff
@@ -83,4 +83,33 @@ const authGuard: Handle = async ({ event, resolve }) => {
 	return resolve(event);
 };
 
-export const handle: Handle = sequence(supabase, authGuard);
+const matchFQDN: Handle = async ({ event, resolve }) => {
+	const host = event.request.headers.get('host');
+	if (!host) {
+		return resolve(event);
+	}
+	const fqdn = host?.split(':')[0];
+	event.locals.fqdn = fqdn;
+	const pathname = event.url.pathname;
+
+	const { data: publicDomain, error } = await event.locals.supabase
+		.from('instance_settings')
+		.select('value')
+		.eq('key', 'domain')
+		.single();
+
+	if (error || !publicDomain) {
+		return resolve(event);
+	}
+
+	if (
+		(pathname.startsWith('/dashboard') || pathname.startsWith('/auth')) &&
+		fqdn !== publicDomain.value
+	) {
+		// To access dashboard we need to go on main instance.
+		redirect(303, `${PUBLIC_URL}${pathname}`);
```

Comment:
```suggestion
		redirect(303, `${publicDomain.value}${pathname}`);
```

Du coup on peut aller jusqu'au bout et enlever la var d'env